ledger/src/blockstore_processor.rs

```diff
@@ -456,29 +468,28 @@ fn rebatch_transactions<'a>(
 fn rebatch_and_execute_batches(
     bank: &Arc<Bank>,
     replay_tx_thread_pool: &ThreadPool,
-    batches: &[TransactionBatchWithIndexes<SanitizedTransaction>],
+    locked_entries: impl ExactSizeIterator<Item = LockedTransactionsWithIndexes<SanitizedTransaction>>,
     transaction_status_sender: Option<&TransactionStatusSender>,
     replay_vote_sender: Option<&ReplayVoteSender>,
     timing: &mut BatchExecutionTiming,
     log_messages_bytes_limit: Option<usize>,
     prioritization_fee_cache: &PrioritizationFeeCache,
 ) -> Result<()> {
-    if batches.is_empty() {
+    if locked_entries.len() == 0 {
         return Ok(());
     }
 
-    let ((lock_results, sanitized_txs), transaction_indexes): ((Vec<_>, Vec<_>), Vec<_>) = batches
-        .iter()
-        .flat_map(|batch| {
-            batch
-                .batch
-                .lock_results()
-                .iter()
-                .cloned()
-                .zip(batch.batch.sanitized_transactions().to_vec())
-                .zip(batch.transaction_indexes.to_vec())
-        })
-        .unzip();
+    // Flatten the locked entries.
+    let ((lock_results, sanitized_txs), transaction_indexes): ((Vec<_>, Vec<_>), Vec<_>) =
+        locked_entries
+            .flat_map(|locked_entry| {
```

Comment:
nit: i prefer destrucutring LockedTransactionsWithIndexes here to ensure its all fields are consumed: i.e.:

```rust
.flat_map(|locked_entry| {
    let LockedTransactionsWithIndexes{lock_results, transactions, starting_index} = locked_entry;
```

Reply:
aa69233e93282705c7e31d899a37fa2907e79244

Reply:
did it slightly differently without the separate let